assets/app/scripts/services/data.js

```diff
@@ -504,14 +504,65 @@ angular.module('openshiftConsole')
     }
   };
 
-  DataService.prototype._watchWebsocketRetries = function(type, context, retry) {
+  DataService.prototype._addWebsocketEvent = function(type, context, eventType) {
     var key = this._uniqueKeyForTypeContext(type, context);
-    if (retry === undefined) {
-      return this._watchWebsocketRetriesMap[key];
+    var events = this._websocketEventsMap[key];
+    if (!events) {
+      events = this._websocketEventsMap[key] = [];
     }
-    else {
-      this._watchWebsocketRetriesMap[key] = retry;
+
+    // Add the event to the end of the array with the time in millis.
+    events.push({
+      type: eventType,
+      time: Date.now()
+    });
+
+    // Only keep 10 events. Shift the array to make room for the new event.
+    var maxWebsocketEvents = 10;
```

Comment:
define the constant outside the function where it can be shared by `isTooManyRecentEvents`